rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+                try:
+                    asset = asset_from_bybit(coin_data['coin'])
+                    amount = deserialize_fval(coin_data['walletBalance'], name=f'Bybit wallet balance for {asset}', location='bybit')  # noqa: E501
+                    if coin_data['usdValue'] != '':
+                        usd_value = deserialize_fval(coin_data['usdValue'], name=f'Bybit usd value for {asset}', location='bybit')  # we don't need to calculate it since it is provided by bybit  # noqa: E501
+                    else:
+                        usd_price = Inquirer().find_usd_price(asset=asset)
+                        usd_value = usd_price * amount
+                except UnknownAsset as e:
+                    self.msg_aggregator.add_warning(
+                        f'Found Bybit balance result with unknown asset '
+                        f'{e.identifier}. Ignoring it.',
+                    )
+                    continue
+                except (DeserializationError, KeyError) as e:
+                    msg = str(e)
+                    if isinstance(e, KeyError):
+                        msg = f'Missing key entry for {msg}.'
+                    return None, f'Error processing Bybit balance entry {coin_data}. {msg}'
+
+                assets_balance[asset] += Balance(amount=amount, usd_value=usd_value)
+
+        return assets_balance, ''
+
+    def _query_deposits(self, options: dict[str, Any]) -> list[AssetMovement]:
+        """
+        Process deposits from bybit. If any asset is unknown or we fail to deserialize an entry
+        the error is logged and we skip the entry.
+        """
+        log.debug(f'querying online deposits for {options}')
+        raw_data = self._paginated_api_query(
+            endpoint='asset/deposit/query-record',
+            options=options,
+        )
+
+        movements = []
+        for movement in raw_data:
+            try:
+                coin = asset_from_bybit(movement['coin'])
+            except UnknownAsset:
+                log.error(f'Could not read assets from trade pair {movement["coin"]}')
+                continue
+
+            try:
+                movements.append(AssetMovement(
+                    timestamp=ts_ms_to_sec(TimestampMS(int(movement['successAt']))),
+                    location=Location.BYBIT,
+                    category=AssetMovementCategory.DEPOSIT,
+                    address=None,
+                    transaction_id=movement['txID'],
+                    asset=coin,
+                    amount=deserialize_asset_amount(movement['amount']),
+                    fee_asset=coin,
+                    fee=Fee(ZERO),
+                    link=movement['txID'],
+                ))
+            except DeserializationError as e:
+                log.error(f'{e} when reading movement for bybit trade {movement}')
+                continue
+
+        return movements
+
+    def _query_withdrawals(self, options: dict[str, Any]) -> list[AssetMovement]:
```

Comment:
This is copy pasta. There is zero season to have a different function for this. Just make one parametrizable function.